### ISSUE TITLE
Number of <seg> in the sequence

### ISSUE THREAD

**Issue by jwh97nn** [dvlab-research/LISA#6]
Hi, I'm curious about the number of \<seg\> that can be predicted in a sequence. 
Based on the test demo below, it appears that only one \<seg\> can be present in the sequence. 

![image](https://github.com/dvlab-research/LISA/assets/16673992/daca60bc-ab2f-488f-a531-c3b5d4e68345)

And can LISA segment multiple instances in a single round?

**Comment by X-Lai** [comment]
Thanks for your interest in our work. Here, we do not differentiate between a single instance or multiple instances. Both of them are represented by a single binary mask. This case is a failure case since the model only segments part of the target objects, and it should have segmented all of them.

**Comment by jwh97nn** [comment]
Thank you for your reply.
So there are no constraints on the number of <seg> tags that can appear in the output sequence?

**Comment by X-Lai** [comment]
Yes, we do not constrain it specially. But all training data contains only one <SEG> in the answer.

**Comment by SiyuanHuang95** [comment]
I would like to have a follow-up question:

How would it extend to multiple <seg> tokens in the sequence? What's the biggest problem to solve? Assuming that we could have more in the answer for the training data. @X-Lai

**Comment by X-Lai** [comment]
The biggest challenge would be the lack of such training data. If we could have enough training data and computing resources, then we could naturally support multiple tokens in the answer (just like what instance segmentation does)

**Comment by SiyuanHuang95** [comment]
Thanks for your quick reply. 

I remembered that the ADE20K contains originally multiple masks for a single image. And the PartImageNet also has parts annotation for a single object. In view of this, was it hard to append multiple <seg> labels in the single sentence?  Or the biggest annotation would be in the construction of ReasonSeg?

**Comment by X-Lai** [comment]
Yes, you're correct. There are already many publicly available instance segmentation datasets that could be used for training. But with regard to samples involving complex reasoning, it takes much more human labor than the current setting.

**Comment by SiyuanHuang95** [comment]
Great thanks for your quick and informative reply. Quite inspiring. 

Then, what do you think would be the part that takes most human labor? Injecting reasoning process or designing task settings that involve multiple <seg>?

BTW, it can be imaged that the training process on multiple <seg> is also difficult since the number is varied.
